posthog-android/src/main/java/com/posthog/android/replay/PostHogReplayIntegration.kt

```diff
@@ -269,12 +274,14 @@ public class PostHogReplayIntegration(
             mouseInteractions.add(mouseInteraction)
         }
 
-        if (mouseInteractions.isNotEmpty()) {
+        if (mouseInteractions.isNotEmpty() && sessionLongerThanMinDuration()) {
             // TODO: we can probably batch those
             // if we batch them, we need to be aware that the order of the events matters
             // also because if we send a mouse interaction later, it might be attached to the wrong
             // screen
+            config.logger.log("Session replay mouse events captured: " + mouseInteractions.size)
             mouseInteractions.capture()
+            mouseInteractions.clear()
```

Comment:
I think we will face a problem here that is:
If a mouse interaction is captured but is not within the min duration and no other interaction is captured after the min duration, the interactions before the min duration are never going to be sent, so we have to call `mouseInteractions.capture()` somewhere as well to be sure that the list is flushed, eg in the `generateSnapshot` method as well.
Most likely we'd need a lock since `generateSnapshot` is executed on a background thread and `generateMouseInteractions` as well so mutations can't race.

Reply:
Ah. Good catch. Let me figure something out here.

Reply:
Have taken a different approach here, where on uninstall I am flushing the events if the threshold was crossed.